docs/book/src/reference/compiler_intrinsics.md

```diff
@@ -372,3 +374,37 @@ Runtime bound checks are not generated, and must be done manually when and where
 
 - `item` is a reference to an array or a reference to a slice;
 - when `index` is a literal, it must be smaller than `item` length;
+
+---
+
+```sway
+__dbg<T>(value: T) -> T where T: Debug
+```
+
+**Description:** Automatically calls the `Debug` trait on the passed `value`, with file, line and column information. The passed value is returned without any modification, allowing `__dbg(...)` to be used inside of any expression.
+
+The code generated by this intrinsic function varies with the compilation mode. For example:
+
+```terminal
+forc build            <- will print everything as expected
+forc build --release  <- nothing will be printed
+```
+
+To enable code generation even on `Release` builds, the flag `force-dbg-in-release` needs to be enabled inside `forc.toml`.
+Example:
+
+```toml
+[project]
+authors = ["Fuel Labs <contact@fuel.sh>"]
+license = "Apache-2.0"
+entry = "main.sw"
+name = "some-project"
+force-dbg-in-release = true
+```
+
+It is strongly suggested to always remove this flag before publishing binaries as it will not have any effect when running
+on real nodes and it only increases gas usage.
```

Comment:
Wondering whether we should even charge gas for debug since this doesnt affect real nodes.
I can imagine scenario's where one has a ton of debug's in code and also wants to gas-golf; but then they'll need to remove/comment out all these lines.

Reply:
Actually `__dbg` will always be removed in release unless `force_dbg_in_release` is set. And for that flag, I expect it to be set only once either in project or workspace `Forc.toml` file. So the removal is actually removal of that one line in `Forc.toml` file.

Reply:
Ah makes sense yep; the fact escaped me that this line is stripped out entirely; so obviously there will be no gas used for this.